### PR TITLE
Regex parser fix

### PR DESCRIPTION
**Pull request by MichalHorky**
This PR fixes bugs in the re2 parser

### CODE REVIEW

File: src/tests-re2parser.cc [hunk 1]
```diff
@@ -836,222 +836,6 @@ TEST_CASE("Mata::RE2Parser basic_parsing")
         CHECK(is_in_lang(aut, Word{'b', 'i', 'd'}));
     }
 
-    SECTION("[ab](c|d)") {
```
Comment (Adda0) [inline src/tests-re2parser.cc:839]: Is there a reason you have removed these additional tests? A failed pull of most recent commits, perhaps?

**Review by Adda0 — APPROVED** [review]
LGTM. Thank you for such quick fixes. Other than the test removal, I can see no issues with the PR. When the tests are resolved, I will merge this PR.

File: src/tests-re2parser.cc (hunk 1 above)
Comment (MichalHorky) [reply src/tests-re2parser.cc]: Yes, that is my mistake. Pull is saying that the branch is up to date with regex_parser_star_iteration, so I'm not sure what was wrong. I added the missing test manually.

Comment (Adda0) [reply src/tests-re2parser.cc]: It is up-to-date at the point of pulling (before your first commit on top of this branch), but the first commit then modifies the test file (deletes these test). Anyway, it looks good now. Thank you for the update.

**Comment by Adda0** [comment]
I am content with the state of this PR. Thank you for yet another bug fix. Merging the PR now.

**Pull request merged**